test/unit/mdc-top-app-bar/foundation.test.js

```diff
@@ -0,0 +1,69 @@
+/**
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {assert} from 'chai';
+import td from 'testdouble';
+
+import {verifyDefaultAdapter} from '../helpers/foundation';
+import MDCTopAppBarFoundation from '../../../packages/mdc-top-app-bar/foundation';
+import {strings} from '../../../packages/mdc-top-app-bar/constants';
+
+
+suite('MDCTopAppBarFoundation');
+
+test('exports strings', () => {
+  assert.isOk('strings' in MDCTopAppBarFoundation);
+});
+
+test('defaultAdapter returns a complete adapter implementation', () => {
+  verifyDefaultAdapter(MDCTopAppBarFoundation, [
+    'hasClass', 'addClass', 'removeClass', 'registerNavigationIconInteractionHandler',
+    'deregisterNavigationIconInteractionHandler', 'notifyNavigationIconClicked',
+  ]);
+});
+
+const setupTest = () => {
+  const mockAdapter = td.object(MDCTopAppBarFoundation.defaultAdapter);
+
+  const foundation = new MDCTopAppBarFoundation(mockAdapter);
+  return {foundation, mockAdapter};
+};
+
+test('#init calls component event registrations', () => {
+  const {foundation} = setupTest();
+
+  foundation.init();
+});
+
+test('foundation returns strings', () => {
```

Comment:
Why not combine this with the "exports strings" test? e.g. https://github.com/material-components/material-components-web/blob/1cc5dd5d3387286980d0510b1333953061b5af01/test/unit/mdc-checkbox/foundation.test.js#L100